### ISSUE TITLE
Feature request: Ignore some lines in elm-review

### ISSUE THREAD

**Issue by girishso** [jfmengels/elm-review#32]
It would be helpful if we can ignore some lines in elm-review.. similar to eslint

```js
// eslint-disable-next-line no-use-before-define
var thing = new Thing();
```

**Comment by jfmengels** [comment]
Hi @girishso 👋
Thanks for opening an issue.
I wrote about this at the bottom of the Readme: https://package.elm-lang.org/packages/jfmengels/elm-review/latest/#is-there-a-way-to-ignore-an-error-or-disable-a-rule-only-in-some-locations-

Please let me know why you would think this would be useful. Do you have specific use-cases in mind (if so, what are they?), or is it more "ESLint has it, so I think it would also be useful here"?

**Comment by girishso** [comment]
Hi @jfmengels !

Thanks for the response, I should have read the Readme first!

Please let me know if our use case makes sense to you.

In our case, we have a 170K LOC Elm codebase.  As you can imagine, we have developed several internal "Libraries" or wrappers with helper functions, for example a wrapper around `Cmd` or a Library for showing Html `table` or dropdowns etc.

Sometimes, we add functions to these wrappers or Libraries that we think will be useful in future, but can't use them immediately mostly due to time pressure.  Just as an example, I recently added `infoNotification`, `errorNotification`, `warnNotification` functions to our `Cmd` wrapper.

```elm
infoNotification : String -> Job msg
infoNotification =
    Toast.info >> addNotification
```

Till now, whenever we needed to display a notification, we used to create the `Toast` then call `addNotification`.  Now with `infoNotification` the two are combined at one place.  Now I used `infoNotification` in the code, but realized we also need `warnNotification` and `errorNotification` functions. And these two later functions are flagged by `elm-review` as unused.

We would like to keep these functions in the wrapper for _discoverability_.

As an interim solution, we've decided leave these functions commented out, with an explanation.

tldr; We need `elm-review` to ignore internal Library or wrapper functions, which practically can't be used immediately.

**Comment by jfmengels** [comment]
I have encountered this kind of situation several times and it's always tricky because it's on the edge of "it could be worthwile to keep" and "we don't need it right now".

I would have several suggestions here, each one being potentially applicable depending on your context

1. Remove those functions entirely. If they are pretty simple, then they are trivial to add anew. If they're complicated, then maybe they are implemented in a way that won't actually be what you want in the feature.


2. Use them at least once somewhere, maybe even just in a test (since the `elm-review-unused` rules count something as used even if it only used in tests).


3.  Temporarily ignore this one file for `NoUnused.Exports` like `config = [ NoUnused.Exports.rule |> Rule.ignoreErrorsForFiles ["src/this-one-file.elm"]`, and add a comment explaining why this exception was made, when and by who. That way, you or someone else can come back later and solve the problem. It will ignore other errors, but `NoUnused.Exports` is a reasonably "low-risk" rule to add exceptions for. One reason why I have rules that look at one single thing is for situations like this, where you can stop one rule temporarily, and then later on, when you're ready, enable it and let the snowball effect roll again. Also, all the exceptions are present in the `ReviewConfig.elm` file, so when you want to start cleanup, you only have to look there.

Anyway, there are several solutions that are pretty reasonable. Later on we will [have even more options](https://github.com/jfmengels/elm-review-design-discussions/issues/4). Even if it is **very** tempting (even to myself!), I think that having comments or directives to disable rules locally are the right solution, and would create more problems than it solves (a very different sort of problem though).

I will therefore close this issue because I don't think I'll implement it, but if you have more feedback/opinions on this, feel free to comment or open a new issue, I appreciate the input!


End note: The rule itself could also add exceptions to this. You could very well clone the rule in your review configuration and either manually add an exception for these functions, or have it take a list of exceptions and specify those in your `ReviewConfig`. I wouldn't recommend going for this solution, but just know it is possible (and maybe appropriate for other rules).
